### ISSUE TITLE
Pioneer Build Error

### ISSUE THREAD

**Issue by mnaamani** [Joystream/joystream#511]
Strangely getting a build error locally, whereas github workflow and netlify checks are passing.

`yarn workspace pioneer build`

on development branch

```
Version 3.7.2
joy-proposals/src/Proposal/ChooseProposalType.tsx:42:55 - error TS2339: Property 'category' does not exist on type 'never'.

42             .filter(typeInfo => !category || typeInfo.category === category)
                                                         ~~~~~~~~

joy-utils/src/react/context/transport.tsx:5:8 - error TS2613: Module '"/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' has no default export. Did you mean to use 'import { Transport } from "/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' instead?

5 import Transport from "../../transport";
         ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:34:13 - error TS7006: Parameter 'unsubscribe' implicitly has an 'any' type.

34       .then(unsubscribe => {
               ~~~~~~~~~~~

joy-utils/src/react/hooks/useTransport.tsx:2:8 - error TS2613: Module '"/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' has no default export. Did you mean to use 'import { Transport } from "/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' instead?

2 import Transport from "../../transport";
         ~~~~~~~~~


Found 4 errors.
```

```sh
MokoBook:joystream mokhtar$ yarn --version
1.16.0
MokoBook:joystream mokhtar$ node --version
v12.17.0
```

**Comment by mnaamani** [comment]
Attempting to fix in #514 Brings up a few other errors..

```
joy-proposals/src/forms/EvictStorageProviderForm.tsx:42:93 - error TS2339: Property 'storageProviders' does not exist on type 'Transport'.

42   const [storageProviders /* error */, , loading] = usePromise<AccountId[]>(() => transport.storageProviders.providers(), []);
                                                                                               ~~~~~~~~~~~~~~~~

joy-proposals/src/forms/SetContentWorkingGroupLeadForm.tsx:81:21 - error TS2339: Property 'council' does not exist on type 'Transport'.

81     () => transport.council.membersExceptCouncil(),
                       ~~~~~~~

joy-proposals/src/forms/SetContentWorkingGroupLeadForm.tsx:85:21 - error TS2339: Property 'contentWorkingGroup' does not exist on type 'Transport'.

85     () => transport.contentWorkingGroup.currentLead(),
                       ~~~~~~~~~~~~~~~~~~~

joy-proposals/src/forms/SetContentWorkingGroupMintCapForm.tsx:9:79 - error TS2339: Property 'contentWorkingGroup' does not exist on type 'Transport'.

9   const [ mintCapacity, error, loading ] = usePromise<number>(() => transport.contentWorkingGroup.currentMintCap(), 0);
                                                                                ~~~~~~~~~~~~~~~~~~~

joy-proposals/src/forms/SetCouncilParamsForm.tsx:72:99 - error TS2339: Property 'council' does not exist on type 'Transport'.

72   const [ councilParams, error, loading ] = usePromise<ElectionParameters | null>(() => transport.council.electionParameters(), null);
                                                                                                     ~~~~~~~

joy-proposals/src/forms/SetMaxValidatorCountForm.tsx:36:63 - error TS2339: Property 'validators' does not exist on type 'Transport'.

36   const [validatorCount] = usePromise<number>(() => transport.validators.maxCount(), NaN);
                                                                 ~~~~~~~~~~

joy-proposals/src/forms/SetStorageRoleParamsForm.tsx:82:78 - error TS2339: Property 'storageProviders' does not exist on type 'Transport'.

82   const [params] = usePromise<IStorageRoleParameters | null>(() => transport.storageProviders.roleParameters(), null);
                                                                                ~~~~~~~~~~~~~~~~

joy-proposals/src/Proposal/Body.tsx:47:21 - error TS2339: Property 'members' does not exist on type 'Transport'.

47     () => transport.members.memberProfile(memberId),
                       ~~~~~~~

joy-proposals/src/Proposal/ChooseProposalType.tsx:23:70 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

23   const [proposalTypes, error, loading] = usePromise(() => transport.proposals.proposalsTypesParameters(), []);
                                                                        ~~~~~~~~~

joy-proposals/src/Proposal/ChooseProposalType.tsx:42:55 - error TS2339: Property 'category' does not exist on type 'never'.

42             .filter(typeInfo => !category || typeInfo.category === category)
                                                         ~~~~~~~~

joy-proposals/src/Proposal/ProposalPreviewList.tsx:55:84 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

55   const [proposals, error, loading] = usePromise<ParsedProposal[]>(() => transport.proposals.proposals(), []);
                                                                                      ~~~~~~~~~

joy-proposals/src/Proposal/VotingSection.tsx:58:21 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

58     () => transport.proposals.voteByProposalAndMember(proposalId, memberId),
                       ~~~~~~~~~

joy-utils/src/react/context/transport.tsx:5:8 - error TS2613: Module '"/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' has no default export. Did you mean to use 'import { Transport } from "/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' instead?

5 import Transport from "../../transport";
         ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:14:21 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

14     () => transport.proposals.proposalById(id),
                       ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:19:21 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

19     () => transport.proposals.votes(id),
                       ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:33:15 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

33     transport.proposals.subscribeProposal(id, refreshProposalData)
                 ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:34:13 - error TS7006: Parameter 'unsubscribe' implicitly has an 'any' type.

34       .then(unsubscribe => {
               ~~~~~~~~~~~


Found 17 errors.
```

**Comment by Lezek123** [comment]
Fixed in: https://github.com/Joystream/joystream/pull/514 (which also introduced the CI to prevent such errors in the future)
